### PR TITLE
Guava checkNotNull to setters added

### PR DESCRIPTION
**Pull request by JuanPoliWolox**
**Summary**

- checkNotNull precondition added to all setters methods

**Trello Cards**

- https://trello.com/c/qb9CggYd/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

**Screenshots**

Captura de Pantalla 2021-10-14 a la(s) 09 47 57

### CODE REVIEW

File: src/main/java/wolox/training/constants/ErrorConstants.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package wolox.training.constants;
+
+public final class ErrorConstants {
+
+    public static final String nullMessage = "Please check the Object supplied, its null!";
```
Comment (juandbc) [inline src/main/java/wolox/training/constants/ErrorConstants.java]: This message can be improved. The word ** object ** can be replaced with a placeholder which is then replaced with the name of the field using `String.format()`.

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -101,38 +104,47 @@ public String getBookIsbn() {
     }
 
     public void setBookGenre(String bookGenre) {
+        Preconditions.checkNotNull(bookGenre, nullMessage);
```
Comment (juandbc) [inline src/main/java/wolox/training/models/Book.java]: This is not a required field
```suggestion
```

File: src/main/java/wolox/training/models/User.java [hunk 3]
```diff
@@ -85,22 +88,27 @@ public LocalDate getUserBirthday() {
...
         this.userUsername = userUsername;
     }
 
     public void setUserName(String userName) {
+        Preconditions.checkNotNull(userName, nullMessage);
         this.userName = userName;
     }
 
     public void setUserBirthday(LocalDate userBirthday) {
+        Preconditions.checkNotNull(userBirthday, nullMessage);
```
Comment (juandbc) [inline src/main/java/wolox/training/models/User.java]: This is not a required field
```suggestion
```

**Review by juandbc — CHANGES_REQUESTED** [review]

**Review by juandbc — APPROVED** [review]

**Review by torsello — APPROVED** [review]
LGTM

**Pull request closed without merging**